lib/pbench/test/functional/server/test_connect.py

```diff
@@ -17,6 +17,11 @@ def test_connect(self, server_client: PbenchServerClient):
         assert "identification" in endpoints
         assert "uri" in endpoints
 
+        # verify all the required authentication fields are present
+        auth_fields = ["issuer", "client", "realm"]
+        assert all(e in auth_fields for e in endpoints["authentication"].keys())
+        assert all(e in endpoints["authentication"].keys() for e in auth_fields)
```

Comment:
Are you sure that you want this to fail if `endpoints["authentication"].keys()` contains values other than what is in `auth_fields`?  (I.e., we have minimum expectations which the server must meet, but do we really want the test to fail if it exceeds them?)

I'm not sure what PyTest will report when one of these assertions fails, but I suspect it won't be very illuminating.  If that's OK with you, then I would write this code this way and save an operation:
```
        auth_fields = {"issuer", "client", "realm"}
        assert set(endpoints["authentication"].keys()) == auth_fields
```

Reply:
> Mr. Jenkins disagrees...and he and I seem to have similar objections.

I too have a similar objection. I shouldn't be doing this assertion `all(e in auth_fields for e in endpoints["authentication"].keys())` 

> we have minimum expectations which the server must meet, but do we really want the test to fail if it exceeds them

That's right, we don't need to fail here.